mlir/test/fusion/e2e/reductions/gfx90a/rock-gemm-reduce-case2.mlir

```diff
@@ -0,0 +1,47 @@
+// This test checks a case where reduction is happening in a middle dimension
+
+// RUN: cat %s | rocmlir-gen -ph -print-results -fut test_reduce -verifier clone - | rocmlir-driver -host-pipeline xmodel -kernel-pipeline full | xmir-runner --shared-libs=%linalg_test_lib_dir/libmlir_rocm_runtime%shlibext,%conv_validation_wrapper_library_dir/libconv-validation-wrappers%shlibext,%linalg_test_lib_dir/libmlir_runner_utils%shlibext,%linalg_test_lib_dir/libmlir_c_runner_utils%shlibext --entry-point-result=void | FileCheck %s --check-prefix=CLONE
```

Comment:
Useless use of `cat`, either use `<` or pass in `%s` as an argument to rocmlir-gen